components/service/glean/src/main/java/mozilla/components/service/glean/storages/EventsStorageEngine.kt

```diff
@@ -12,8 +14,15 @@ import org.json.JSONArray
  * This singleton handles the in-memory storage logic for events. It is meant to be used by
  * the Specific Events API and the ping assembling objects. No validation on the stored data
  * is performed at this point: validation must be performed by the Specific Events API.
+ *
+ * This class contains a reference to the Android application Context. While the IDE warns
+ * us that this could leak, the application context lives as long as the application and this
+ * object. For this reason, we should be safe to suppress the IDE warning.
  */
+@SuppressLint("StaticFieldLeak")
 internal object EventsStorageEngine : StorageEngine {
+    override lateinit var applicationContext: Context
```

Comment:
Having the `applicationContext` as lateinit here in the Singleton and then assigning it manually from the manager is a bit confusing to follow. See comment in the manager class.

Reply:
Yeah, it's a bit weird :\ I wish there was something like "getApplicationContext()" somewhere